qiskit_dynamics/pulse_simulator/pulse_simulator.py

```diff
@@ -0,0 +1,526 @@
+        rng = np.random.default_rng(self.options.seed_simulator)
+        experiment_results = []
+        for (
+            experiment_name,
+            solver_result,
+            measurement_subsystems,
+            memory_slot_indices,
+            num_memory_slots,
+        ) in zip(
+            experiment_names,
+            solver_results,
+            measurement_subsystems_list,
+            memory_slot_indices_list,
+            num_memory_slots_list,
+        ):
+            experiment_results.append(
+                self.options.experiment_result_function(
+                    experiment_name,
+                    solver_result,
+                    measurement_subsystems,
+                    memory_slot_indices,
+                    num_memory_slots,
+                    self,
+                    seed=rng.integers(low=0, high=9223372036854775807),
+                )
+            )
+
+        # Construct full result object
+        return Result(
+            backend_name=self.name,
+            backend_version=self.backend_version,
+            qobj_id="",
+            job_id=job_id,
+            success=True,
+            results=experiment_results,
+            date=datetime.datetime.now().isoformat(),
+        )
+
+    @property
+    def max_circuits(self):
+        return None
+
+    @property
+    def target(self) -> Target:
+        return self._target
+
+    @property
+    def meas_map(self) -> List[List[int]]:
+        return self.options.meas_map
+
+
+def default_experiment_result_function(
+    experiment_name: str,
+    solver_result: OdeResult,
+    measurement_subsystems: List[int],
+    memory_slot_indices: List[int],
+    num_memory_slots: Union[None, int],
+    backend: PulseSimulator,
+    seed: Optional[int] = None,
+) -> ExperimentResult:
+    """Default routine for generating ExperimentResult object.
+
+    Transforms state out of rotating frame into lab frame using ``backend.options.solver``,
+    normalizes if ``backend.options.normalize_states==True``, and computes measurement results
+    in the dressed basis based on measurement-related options in ``backend.options`` along with
+    the measurement specification extracted from the experiments, passed as args to this function.
+
+    Args:
+        experiment_name: Name of experiment.
+        solver_result: Result object from :class:`Solver.solve`.
+        measurement_subsystems: Labels of subsystems in the model being measured.
+        memory_slot_indices: Indices of memory slots to store the results in for each subsystem.
+        num_memory_slots: Total number of memory slots in the returned output. If ``None``,
+            ``max(memory_slot_indices)`` will be used.
+        backend: The backend instance that ran the simulation. Various options and properties
+            are utilized.
+        seed: Seed for any random number generation involved (e.g. when computing outcome samples).
+    Returns:
```

Comment:
nitpick: I would add a blank line between Args, Returns, Raises for all doc strings

Reply:
Just added blank lines between these for all new doc strings.